electrum_nmc/electrum/gui/qt/main_window.py

```diff
@@ -3360,9 +3363,23 @@ def update_buy_names_preview(self):
         identifier = self.buy_names_new_name_lineedit.text().encode('ascii')
         identifier_formatted = format_name_identifier(identifier)
         self.buy_names_preview_label.setText(_("Name to register: ") + identifier_formatted)
-
+        
+        # Convert to hexadecimal string
+        hex_text = name_to_str(identifier, Encoding.HEX)
+        self.buy_names_new_name_hex_lineedit.setText(hex_text)
+        
         self.buy_names_status_label.setText(_(""))
         self.buy_names_register_button.hide()
+      
+    def update_register_name_from_hex(self):
+        hex_text = self.buy_names_new_name_hex_lineedit.text()
+        try:
+    	    # Convert the hex text to bytes and decode it as ASCII
+            identifier = name_from_str(hex_text, Encoding.HEX).decode('ascii')
```

Comment:
Use `name_to_str` to convert to ASCII.

Reply:
In order to work with the hex_text variable, which is a string, it is necessary to convert it into a bytes object. This conversion can be achieved using the bytes.fromhex function available in the name_from_str module. The main purpose of using this function is to facilitate the conversion process. Alternatively, if the name_to_str module is used instead, the hex_text would need to be converted to a bytes format using the bfh function. 
Maybe a helper function for formating hex text -> bytes similar to [this example](https://github.com/namecoin/electrum-nmc/blob/master/electrum_nmc/electrum/gui/qt/main_window.py#L3361). 

```
identifier_hex_formatted = bfh(identifier_hex)
identifier = name_to_str(formated, Encoding.ASCII)
```

Reply:
Why not chain `name_from_str` (from hex) with `name_to_str` (to ASCII)?

Reply:
Alright!